tests/src/test/scala/it/unibo/scafi/test/functional/stdlib/TestBlockT.scala

```diff
@@ -0,0 +1,166 @@
+      override def main(): (Int, Int) = (
+        T(manyManyRounds, 0, unitaryDecay),
+        //this can be seen as a round counter
+        rep(0)(_ + 1)
+      )
+    }, ntimes = fewRounds)(net)
+
+    assert(net.valueMap[(Int, Int)]().forall { case (_, (done: Int, todo: Int)) => done + todo == manyManyRounds })
+  }
+
+  Block_T should "support T with unitary decay and custom floor value" in new SimulationContextFixture {
+    val floorValue = 1
+    exec(new TestProgram {
+      override def main(): Int = T(10, floorValue, unitaryDecay)
+    }, ntimes = someRounds)(net)
+
+    assertNetworkValues((0 to 8).zip(List(
+      floorValue, floorValue, floorValue,
+      floorValue, floorValue, floorValue,
+      floorValue, floorValue, floorValue
+    )).toMap)(net)
+  }
+
+  Block_T should "support T with unitary decay and negative floor value" in new SimulationContextFixture {
+    val floorValue: Int = -10
+    exec(new TestProgram {
+      override def main(): Int = T(10, floorValue, unitaryDecay)
+    }, ntimes = someRounds)(net)
+
+    assertNetworkValues((0 to 8).zip(List(
+      floorValue, floorValue, floorValue,
+      floorValue, floorValue, floorValue,
+      floorValue, floorValue, floorValue
+    )).toMap)(net)
+  }
+
+  Block_T should("support timer operation") in new SimulationContextFixture {
+    exec(new TestProgram {
+      override def main(): Int = timer(10)
+    }, ntimes = someRounds)(net)
+
+    assertNetworkValues((0 to 8).zip(List(
+      0, 0, 0,
+      0, 0, 0,
+      0, 0, 0
+    )).toMap)(net)
+  }
+
+  Block_T should("support limitedMemory - without expiration") in new SimulationContextFixture {
+    val value: Int = 10
+    val expValue: Int = -1
+    exec(new TestProgram {
+      override def main(): Int = limitedMemory(value, expValue, manyManyRounds)._1
+    }, ntimes = fewRounds)(net)
+
+    assertNetworkValues((0 to 8).zip(List(
+      value, value, value,
+      value, value, value,
+      value, value, value
+    )).toMap)(net)
+  }
+
+  Block_T should("support limitedMemory - with expiration") in new SimulationContextFixture {
+    val value: Int = 10
+    val expValue: Int = -1
+    exec(new TestProgram {
+      override def main(): Int = limitedMemory(value, expValue, 10)._1
+    }, ntimes = manyRounds)(net)
+
+    assertNetworkValues((0 to 8).zip(List(
+      expValue, expValue, expValue,
+      expValue, expValue, expValue,
+      expValue, expValue, expValue
+    )).toMap)(net)
+  }
+
+  Block_T should("support sharedTimerWithDecay") in new SimulationContextFixture {
+    val maxStdDev: Int = 2
+    exec(new TestProgram {
+      override def main(): Int = sharedTimerWithDecay(1, 1)
+    }, ntimes = manyManyRounds)(net)
+
+    //standard deviation inside the same group should be low
+    assert(stdDev(net.valueMap[Int]().filterKeys(_ != 8).values) < maxStdDev)
+  }
+
+  Block_T should("T should restart after branch switch") in new SimulationContextFixture {
+    var count = 0
```

Comment:
I'd rewrite this test as follows:

- the program should be something as `branch(sense[Boolean]("sensor")){ T(...) }{ someDefaultValue }`
- run some rounds and check the timer hit the floor
- change sensor to `false`, run few rounds
- change sensor back to `true`, run very few rounds (e.g., less than `ceiling-floor`) and check that no device has hit the floor